client/packages/system/src/Manage/Preferences/Components/EditWarningWhenMissingRecentStocktakeData.tsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import {
+  Box,
+  Typography,
+  NumericTextInput,
+  Switch,
+  WarnWhenMissingRecentStocktakeDataNode,
+  InputWithLabelRow,
+  useTranslation,
+} from '@openmsupply-client/common';
+
+// Backend returns snake_case, but TypeScript types expect camelCase
+type WarnWhenMissingRecentStocktakeDataSnakeCase = {
+  enabled: boolean;
+  max_age?: number;
+  min_items?: number;
+  maxAge?: number;
+  minItems?: number;
+};
+
+export const EditWarningWhenMissingRecentStocktakeData = ({
+  value,
+  update,
+  disabled = false,
+}: {
+  value: WarnWhenMissingRecentStocktakeDataNode;
+  update: (
+    value: Omit<WarnWhenMissingRecentStocktakeDataNode, '__typename'>
+  ) => void;
+  disabled?: boolean;
+}) => {
+  const t = useTranslation();
+
+  // The backend returns snake_case (max_age, min_items) but we need to handle both formats
+  const getValue = (val: WarnWhenMissingRecentStocktakeDataSnakeCase) => {
+    const maxAge = val?.max_age ?? val?.maxAge ?? 0;
+    const minItems = val?.min_items ?? val?.minItems ?? 0;
+    const enabled = val?.enabled ?? false;
+
+    return { maxAge, minItems, enabled };
+  };
+
+  const { maxAge, minItems, enabled } = getValue(
+    value as WarnWhenMissingRecentStocktakeDataSnakeCase
+  );
+
+  // Helper to ensure all required fields have valid values (excluding __typename for input)
+  const createUpdateValue = (
+    updates: Partial<Omit<WarnWhenMissingRecentStocktakeDataNode, '__typename'>>
+  ) => {
+    const newValue = {
+      enabled,
+      maxAge,
+      minItems,
+      ...updates,
+    };
+    update(newValue);
+  };
+
+  const handleMaxAgeChange = (newMaxAge?: number | undefined) => {
+    createUpdateValue({ maxAge: newMaxAge });
+  };
+
+  const handleMinItemsChange = (newMinItems?: number | undefined) => {
+    createUpdateValue({ minItems: newMinItems });
+  };
+
+  return (
```

Comment:
As in my last comment, if we're making a specific component anyway then maybe not worth all the effort?

Reply:
Agree, the inner prefs UI should be auto-rendered like the root layer, otherwise not much point have them dynamically queried, right?

Reply:
I think it should be reasonably straightforward to create a generic "ComplexPref" wrapper that renders the inner prefs with the same code as the root prefs?

Reply:
I've tried to capture this idea by creating the `PreferenceAccordion` component, so that can contain what ever groupings or custom components are needed.